lib/sitemap-stream.ts

```diff
@@ -206,7 +207,10 @@ export class SitemapStream extends Transform {
     // Check if the size would be exceeded by the new item
     // and throw if it would exceed (when size limit enabled)
     if (this.byteLimit !== undefined) {
-      if (this._byteCount + itemOutput.length > this.byteLimit) {
+      if (
+        this._byteCount + Buffer.byteLength(itemOutput) >
```

Comment:
Good catch!  We definitely want the byte length here.  We should add some tests to make sure we're using the byte length and not char length.